patch_rebaser/patch_rebaser.py

```diff
@@ -1,3 +1,164 @@
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
-
 """Main module."""
+
+import configparser
+import logging
+import os
+
+from distroinfo import info, query
+from git_wrapper.repo import GitRepo
+
+LOGGER = logging.getLogger("patch_rebaser")
+
+
+def find_patches_branch(repo, remote, distgit_branch):
+    """Guess patches branch name"""
+    # Adapted from rdopkg guess.find_patches_branch
+    parts = distgit_branch.split('-')
+    while parts:
+        if 'trunk' in distgit_branch:
+            branch = '%s-trunk-patches' % '-'.join(parts)
+        else:
+            branch = '%s-patches' % '-'.join(parts)
+        LOGGER.debug("Checking if branch %s exists...", branch)
+        if repo.branch.exists(branch, remote):
+            return branch
+        parts.pop()
+    return None
+
+
+def get_downstream_distgit_branch(user):
+    """Get downstream distgit branch info from DLRN projects.ini"""
+    # projects.ini location is standard
+    DLRN_PROJECTS_INI = "/usr/local/share/dlrn/{0}/projects.ini".format(user)
+
+    config = configparser.ConfigParser()
+    config.read(os.path.expandvars(DLRN_PROJECTS_INI))
```

Comment:
I don't think we need expandvars here. The path being passed in looks fully qualified.

Reply:
Well spotted, I removed it.